system/system.go

```diff
@@ -51,11 +56,83 @@ func (i *Info) Clone() *Info {
 		MessagesDropped:     atomic.LoadInt64(&i.MessagesDropped),
 		Retained:            atomic.LoadInt64(&i.Retained),
 		Inflight:            atomic.LoadInt64(&i.Inflight),
-		InflightDropped:     atomic.LoadInt64(&i.InflightDropped),
-		Subscriptions:       atomic.LoadInt64(&i.Subscriptions),
-		PacketsReceived:     atomic.LoadInt64(&i.PacketsReceived),
-		PacketsSent:         atomic.LoadInt64(&i.PacketsSent),
-		MemoryAlloc:         atomic.LoadInt64(&i.MemoryAlloc),
-		Threads:             atomic.LoadInt64(&i.Threads),
+		//		InflightDropped:     atomic.LoadInt64(&i.InflightDropped),
+		Subscriptions:   atomic.LoadInt64(&i.Subscriptions),
+		PacketsReceived: atomic.LoadInt64(&i.PacketsReceived),
+		PacketsSent:     atomic.LoadInt64(&i.PacketsSent),
+		MemoryAlloc:     atomic.LoadInt64(&i.MemoryAlloc),
+		Threads:         atomic.LoadInt64(&i.Threads),
+	}
+}
+
+func (i *Info) RegisterPrometheusMetrics(registry prometheus.Registerer) {
+	if registry == nil {
+		registry = prometheus.DefaultRegisterer
+	}
+
+	type metrics struct {
+		metricType string
+		name       string
+		help       string
+		value      *int64
 	}
+
+	metricsList := []metrics{
+		{"c", "bytes_received", "A count of total number of bytes received", &i.BytesReceived},
+		{"c", "bytes_sent", "A counter total number of bytes sent", &i.BytesSent},
+		{"g", "clients_connected", "A gauge of number of currently connected clients", &i.ClientsConnected},
+		{"g", "clients_disconnected", "A gauge of total number of persistent clients", &i.ClientsDisconnected},
+		{"c", "clients_maximum", "A count of maximum number of active clients that have been connected", &i.ClientsMaximum},
+		{"g", "clients_total", "A gauge of total number of connected and disconnected clients with a persistent session currently connected and registered", &i.ClientsTotal},
+		{"c", "messages_received", "A counter of total number of publish messages received", &i.MessagesReceived},
+		{"c", "messages_sent", "A counter of total number of publish messages sent", &i.MessagesSent},
+		{"c", "messages_dropped", "A counter of total number of publish messages dropped to slow subscriber", &i.MessagesDropped},
+		{"g", "retained", "A gauge of total number of retained messages active on the broker", &i.Retained},
+		{"g", "inflight", "A gauge of the number of messages currently in-flight", &i.Inflight},
+		//		{"c", "inflight_dropped", "A", &i.InflightDropped},
```

Comment:
As above

Reply:
same as above, InflightDropped is not updated